js/client.js

```diff
@@ -1292,9 +1293,30 @@ class PalaceClient extends PalaceProtocol {
 	}
 
 	goto(url) {
-		var blah = url.trim().replace('palace://','').split(':'); //should use forgiving regex
+		var connectInfo = url.trim().replace('palace://','').split(':'); //should use forgiving regex
 		this.retryRegistration = false;
-		super.connect(blah[0],blah[1]);
+        var xmlHttp = new XMLHttpRequest();
```

Comment:
I am wondering what is the purpose of this logic exactly? I can kinda see some benefits, but it defeats the purpose of the user inputting a fully qualified URI then. If the idea behind this is to reduce IP allocation for hosting providers, there is already a better approach for this on the DNS level: https://en.wikipedia.org/wiki/SRV_record.

An example configuration would be:

~~~zone
_palace._tcp.aperitio.epalaces.com. 300 IN SRV 0 1 9991 server4.epalaces.com.
_palace._tcp.lessfun.epalaces.com.  300 IN SRV 0 1 9992 server4.epalaces.com.
~~~

and the corresponding query would be:

~~~
_palace._tcp.aperitio.epalaces.com. IN SRV
~~~

Doing this on the application level seems a bit fickle and error prone unless we are defensive about it (ie schema validation, value validation and probably other stuff that doesn't come to mind right now).

-----

Further, this can be simplified using `fetch`, and `async`/`await` here is an example (haven't tested it):

~~~javascript
let [ host, port ] = connectInfo
const request = await fetch(`http://${host}/palace.json`)

// Convert to int
port = parseInt(port)

if (request.ok) {
  let serverDesc = await req.json()
  
  if ('host' in serverDesc && 'port' in serverDesc)
    super.connect(serverDesc.host, serverDesc.port)
  else
    console.error('Invalid palace.json returned') // See note[1]
    
} else {
  console.warn(`Server returned '${req.status} ${req.statusText}' when requesting ${req.url}`)
  super.connect(host, port)
}
~~~

If you do decide do use this approach, you'll need to change the function signature to `async goto(uri)`

[1]: Ideally this would be reflected in the UI, not in the console, but I am not familiar enough with the codebase to put it in here right now.

Reply:
I considered the SRV records but as i dont have any clue how they work in realbasic (pchat client) i wasnt sure if using it would be a hassle for backwards compatiblity for pchat v4, i decided on the json approuch.

Reply:
And correct, its to reduce IP allocation.